src/nanosvg.h

```diff
@@ -2634,16 +2650,43 @@ static void nsvg__endElement(void* ud, const char* el)
 		nsvg__popAttr(p);
 	} else if (strcmp(el, "path") == 0) {
 		p->pathFlag = 0;
+		p->shapeFlag = 0;
 	} else if (strcmp(el, "defs") == 0) {
 		p->defsFlag = 0;
+	} else if (strcmp(el, "title") == 0) {
+		p->titleFlag = 0;
+	} else if (strcmp(el, "rect") == 0 ||
+			   strcmp(el, "circle") == 0 ||
+			   strcmp(el, "ellipse") == 0 ||
+			   strcmp(el, "line") == 0 ||
+			   strcmp(el, "polyline") == 0 ||
+			   strcmp(el, "polygon") == 0) {
+		p->shapeFlag = 0;
 	}
 }
 
 static void nsvg__content(void* ud, const char* s)
 {
-	NSVG_NOTUSED(ud);
-	NSVG_NOTUSED(s);
-	// empty
+	NSVGparser* p = (NSVGparser*)ud;
+	if (p->titleFlag) {
+		int len = strlen(s);
+		const int lim = sizeof(((NSVGshape *)(nullptr))->title);
```

Comment:
`nullptr` is C++ (and actually C++11).

Using your own implementation depending on the used programming language might help:
``` C
#ifdef __cplusplus
	#if __cplusplus > 201103L
	// has nullptr
	#define NSVG_NULLPTR nullptr
	#else

// implement nullptr
const class {
public:
	template<class T> operator T*() const { return 0; }
	template<class C, class T> operator T C::*() const { return 0; }
private:
	void operator&() const;
} NSVG_NULLPTR = {};

	#endif
#else //!__cplusplus
	#define NSVG_NULLPTR (void*)0
#endif

//...
const int lim = sizeof(((NSVGshape *)(NSVG_NULLPTR))->title);
```

Reply:
Or better, use a macro for the size of `title`:
``` C
#define NSVGATTRIB_TITLE_SIZE 64

 typedef struct NSVGattrib
 {
 	char id[64];
	char title[NSVGATTRIB_TITLE_SIZE];
// ...

const int lim = NSVGATTRIB_TITLE_SIZE;
```